frontend/src/pages/RoomPage.tsx

```diff
@@ -1,30 +1,61 @@
-import { useEffect, useState } from "react";
+import { useCallback, useEffect, useState } from "react";
 import { useNavigate, useParams } from "react-router-dom";
-import { Avatar, Divider, Grid, Stack, Typography } from "@mui/material";
+import { Avatar, Divider, Grid, Stack } from "@mui/material";
 import { useSnackbar } from "notistack";
 
+import { SOCKET_IO_DISCONNECT_REASION } from "src/constants/socket.io";
+import { useAuth } from "src/contexts/AuthContext";
 import { useSocket } from "src/contexts/WsContext";
+import { useGetUserName } from "src/hooks/useUsers";
+import { nameToInitials } from "src/utils/string";
 
 import { Editor } from "../components/Editor";
 import { Question } from "../components/Question";
 import { StyledButton } from "../components/StyledButton";
 
 import { ROOM_EVENTS, ROOM_NAMESPACE } from "~shared/constants";
+import {
+  JoinedPayload,
+  PartnerDisconnectPayload,
+  PartnerLeavePayload,
+} from "~shared/types/api";
+
+type participant = {
+  userId: number;
+  name?: string;
```

Comment:
Is there a reason why name is an optional field? If there exist a user then i assume there should be a name?

Reply:
The name is not sent by the websocket to us, we have to fetch it separately to retrieve the partner's name.